Sources/ProjectDescription/Dependency.swift

```diff
@@ -26,22 +26,47 @@ public enum Dependency: Codable, Equatable {
         case revision(String)
     }
 
+    /// Requirement for the Swift Package Manager dependency.
+    public enum SPMRequirement: Codable, Equatable {
+        /// Mimics `.exact("1.2.3")` from `Package.swift`.
+        case exact(Version)
+        /// Mimics `.upToNextMajor(from: "1.2.3")` from `Package.swift`.
+        case upToNextMajor(Version)
+        /// Mimics `.upToNextMinor(from: "1.2.3")` from `Package.swift`.
+        case upToNextMinor(Version)
+        /// Mimics `from: "1.2.3"` from `Package.swift`.
+        case atLeast(Version)
+        /// Mimics `"1.2.3"..<"1.2.6"` from `Package.swift`.
+        case range(Range<Version>)
+        /// Mimics `1.2.3"..."1.2.6` from `Package.swift`.`
+        case closedRange(ClosedRange<Version>)
```

Comment:
I think `SPMRequirement` should more closely resemble `Package.Dependency.Requirement`. Having `atLeast`, `range` and `closedRange` cases does not really feel right to me (though I admit it's subjective). I'd either revise `spm` case to contain a different parameter than `requirement` (as in `Package.swift`) or instead of `atLeast`, `range`, and `closedRange` add only `range(from: Version, to: Version)` and `range(from: Version)`. This is also more similar to how it's done currently in `ProjectDescription.Package`

Reply:
Agree with Marke, I'd reuse the same requirement that is being used for the current support for Swift Package Manager. @ollieatkinson did the implementation and he made sure that the implementation was aligned with the API that Apple provides.